### PR TITLE
Replaces the changeling gamemode with a vampires + changelings gamemode.

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

Title + Adds extra logic to gamemodes that prevents someone from being picked for multiple antagonist roles.
## Why It's Good For The Game
<details>
<summary>
I've gotten on my soapbox on the Discord a few times already, so I'll copy+paste my thoughts from there.
</summary>

I think the changeling gamemode is very flawed, and here are the main reasons why.

- A roundtype of exclusively KOS antags means escalation will only ever increase.

- Changelings don't have enough varied playstyles to make 5-7 in a single round interesting.

- Having such relatively few antags in a round that perma-die when dealt with means that good security officers can quickly run out of things to do, and will cryo with the meta-knowledge that the round has like 2 interesting clings left.

- Once one changeling goes loud, every other changeling is subject to whatever the highest level of escalation so far has been. I feel like it's very uncommon that processing gibbers ever get disassembled, even if the main problem cling has been dealt with.

- Playing changeling stealthily tends to result in people getting round removed very early. Since a stealthy changeling having their identity leaked is the worst thing that could happen to them, they're incentivized to perma-kill both victims AND witnesses.

- Changelings are nearly unkillable if there's a team of more than two, or security is too spread out by multiple loud clings. Having less changelings in a round will make this happen significantly less.

Overall, I strongly believe that changelings work significantly better when there are other antagonists in the round, both for security, and the changeling's sake.
</details>

## Testing

Made sure that I couldn't get two different antagonists at once, and used breakpoints to simulate what a full population round might look like.
<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>
image

## Changelog

:cl:
del: Removed the changeling gamemode from rotation.
add: Added the vampires + changelings gamemode to rotation.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/vampire/vampire_chan.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/datum/game_mode/vampire/changeling
+	name = "vampire_changeling"
+	config_tag = "vampchan"
+	restricted_jobs = list("AI", "Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Blueshield", "Nanotrasen Representative", "Magistrate", "Chaplain", "Internal Affairs Agent", "Nanotrasen Navy Officer", "Special Operations Officer", "Syndicate Officer", "Solar Federation General")
+	protected_species = list("Machine")
+	required_players = 1 //Testing, TODO blah balh
```
Comment (lewcc) [inline code/game/gamemodes/vampire/vampire_chan.dm:7]: You know the drill

File: code/game/gamemodes/vampire/vampire_chan.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			break
+		var/datum/mind/changeling = pick_n_take(possible_changelings)
+		pre_changelings += changeling
+		changeling.restricted_roles = (restricted_jobs + secondary_restricted_jobs)
+		changeling.special_role = SPECIAL_ROLE_CHANGELING
+
+	return ..()
+
+/datum/game_mode/vampire/announce()
+	to_chat(world, "<B>The current game mode is - Vampires+Changelings!</B>")
```
Comment (lewcc) [inline code/game/gamemodes/vampire/vampire_chan.dm:40]:
```suggestion
	to_chat(world, "<b>The current game mode is - Vampires+Changelings!</b>")
```

File: code/game/gamemodes/vampire/vampire_chan.dm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	to_chat(world, "<B>Some of your co-workers are vampires, and others might not even be your coworkers!</B>")
```
Comment (lewcc) [inline code/game/gamemodes/vampire/vampire_chan.dm:41]:
```suggestion
	to_chat(world, "<b>Some of your co-workers are vampires, and others might not even be your coworkers!</b>")
```

File: code/game/gamemodes/vampire/vampire_chan.dm (hunk 1 above)
Comment (BiancaWilkson) [reply code/game/gamemodes/vampire/vampire_chan.dm:7]: This is exactly why I write these comments, thanks for catching lol

File: code/game/gamemodes/vampire/vampire_gamemode.dm [hunk 4]
```diff
@@ -22,7 +24,7 @@
 
 	var/list/datum/mind/possible_vampires = get_players_for_role(ROLE_VAMPIRE)
 
-	var/vampire_amount = 1 + round(num_players() / 10)
+	var/vampire_amount = 1 + (round(num_players() / 10) * (1 - vampire_penalty))
```
Comment (DGamerL) [inline code/game/gamemodes/vampire/vampire_gamemode.dm:27]: Shouldn't this be just `* vampire_penalty`

**Review by DGamerL — COMMENTED** [review]
The cling ratio might not be a lot tbh, at 100 players in-round it'll only be 3 clings to 6 vamps, whereas normally it'd be 11 vamps

**Comment by chuga-git** [comment]
> The cling ratio might not be a lot tbh, at 100 players in-round it'll only be 3 clings to 6 vamps, whereas normally it'd be 11 vamps

What are the equivalent 100 player ratios for trifecta, tot-cling and tot-vamp?

File: code/game/gamemodes/vampire/vampire_gamemode.dm (hunk 4 above)
Comment (BiancaWilkson) [reply code/game/gamemodes/vampire/vampire_gamemode.dm:27]: ....It should, yes.

**Comment by BiancaWilkson** [comment]
> > The cling ratio might not be a lot tbh, at 100 players in-round it'll only be 3 clings to 6 vamps, whereas normally it'd be 11 vamps
> 
> What are the equivalent 100 player ratios for trifecta, tot-cling and tot-vamp?

The ratio I used is the same for tot-cling and tot-vamp, but those gamemodes don't actually reduce the number of traitors when changelings/vampires are added, which I think is too important for things as high-impact as both vampires and changelings. I can definitely tweak the values though.

**Comment by DGamerL** [comment]
> > The cling ratio might not be a lot tbh, at 100 players in-round it'll only be 3 clings to 6 vamps, whereas normally it'd be 11 vamps
> 
> What are the equivalent 100 player ratios for trifecta, tot-cling and tot-vamp?

Trifecta, assuming equal distribution, is 6 tots, 3 vamps and 2 clings
Totcling and Totvamp are 20 tots + 3 clings/vamps (Yes this number is stupidly high)

**Comment by 1080pCat** [comment]
> Trifecta, assuming equal distribution, is 6 tots, 3 vamps and 2 clings

>gamemodes called trifecta
>distribution isn't 3's 
stg

**Review by DGamerL — APPROVED** [review]

**Pull request merged**